### ISSUE TITLE
Install Works Half-Way Home.

### ISSUE THREAD

**Issue by pmtremblay** [Xigtun/xps-9570-mojave#9]
Bonjour,

Here's my config of my Dell New XPS 9570
CPU:
Intel Core i9-8950HK (12M Cache, 2.9-X.X GHz)​
GPU:
Intel UHD 630 + NVIDIA® GeForce™ GTX 1050Ti graphics card (4GB GDDR5 video memory)
RAM:
Crucial 32GB (2x16GB) DDR4 2666MHz [2 SoDIMM slots, up to 32GB of Dual Channel DDR4, 2 666 MHz]
Storage:
2TB 3D NAND WD Blue WDS200T2B0B​
Display:
15.6" 4K Ultra HD (3840 x 2160) InfinityEdge Anti-Reflective Touch IPS100% AdobeRGB 400-Nits display​
Audio:
Realtek ALC3266 -​ Waves MaxxAudio® Pro technology, 2 × 2W = 4W total
Wireless:
Dell XPS 15 9550 15.6" Genuine Wireless-AC Wi-Fi Card HHKJD BCM943602BAED(DW1830),NGFF(keyA/E), M.2 2630,2.4,5GHz, 802.11ac, 1300Mbps, Bluetooth:4.1
(I switched the original Killer 1535 for the BCM943602BAED(DW1830),NGFF(keyA/E), M.2 2630,2.4,5GHz, 802.11ac, 1300Mbps, Bluetooth:4.1)
(Intel® Wireless-AC 9260 802.11 ac/a/b/g/n 2x2 NGFF w/ Bluetooth 5.0)
Battery:
3 Cell 56/97WHr (8.2Ah @ 12V)​
Thunderbolt 3 (4 lanes of PCI Express Gen 3):
Intel JHL6240​
Power:
130 Watt 3 Prong AC Adapter with 6 ft Power Cord (will work worldwide, includes FREE plug adapter for those without "U.S. style" outlets

I took the EFI_Install and copied it onto an 16GB USB key along with Mojave Install.

I can get to the first install screen. It installs Mojave OS but it does not resume properly after the first reboot.

Is there something I missed?

Thanks.
PM

**Comment by kevinshroff** [comment]
You need to switch to EFI folder and boot with that instead of EFI_Install, as described in the instructions

**Comment by pmtremblay** [comment]
Thanks. I did that.
1- I first start fresh with the content of the folder named: install_EFI onto EFI on my USB drive.
2- Put the USB key into my XPS mobile and power it up.
3- Press F12 and select the "Installer macOS Mojave" USB drive as my UEFI boot up process.
4- I then reformat the SSD drive to Mac OS Extended (Journaled) with GUID Partition Map and name it "MojaveSSD".
5- Then I start the Mojave Install Process. Everything works perfectly up to here.
6- Then the screen turns black (reboot). This is where I turn off my mobile.
7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.
8- I put the newly updated USB\EFI key into my XPS mobile and power it up.
9- Press F12 and select the "MojaveSSD" SSD drive as I want to continue the install.
10- Then it crashes. I does not continue to install.

Let me know if I do something wrong or if I missed a step.
Thanks.
PM

**Comment by Raymonf** [comment]
Try to set your ig-platform-id to 0x12345678?

**Comment by pmtremblay** [comment]
> You need to switch to EFI folder and boot with that instead of EFI_Install, as described in the instructions
Thanks kevinshroff.

I changed ig-platform-id to 0x12345678 (as suggested by Raymonf), tried your suggestion (use simply the EFI and ignore install_EFI folder) and I went further to the full install of Mojave. Once the installation process finished, I had no control of the mouse so I could not hit the Restart button in the middle of the Install splash window. Nonetheless it rebooted by itself after few seconds and then, I went back to Clover bootloader (F12) and selected MojaveSSD disk but it crashed again.

I'm stuck!
What else should I do?
Many thanks.

**Comment by Raymonf** [comment]
Move the files from CLOVER into your EFI partition's CLOVER folder. You should be set after that. If not, select "boot with injected kexts".

**Comment by pmtremblay** [comment]
Hey Ramonf
In my USB Clover bootloader, I selected "boot with injected kexts" but it crashes when attempting to start with MojaveSSD drive. As for the other suggestion you made, I'm not sure I understand. You mean copying the content of the USB:EFI\CLOVER into my SSD:EFI?
Thanks for your quick response.

PS - I wish the instructions of this project were cristal clear.

**Comment by Raymonf** [comment]
Ah, I see. Boot with 0x12345678 ig-platform-id and then get into the system. Then, install Clover to the ESP. Then, mount your EFI partition and replace the entire Clover and Boot folders with the ones from this repo.

在 2018年10月30日週二 下午1:19，pmtremblay <notifications@github.com<mailto:notifications@github.com>> 寫道：

Hey Ramonf
In my USB Clover bootloader, I selected "boot with injected kexts" but it crashes when attempting to start with MojaveSSD drive. As for the other suggestion you made, I'm not sure I understand. You mean copying the content of the USB:EFI\CLOVER into my SSD:EFI?
Thanks for your quick response.

PS - I wish the instructions of this project were cristal clear.

**Comment by pmtremblay** [comment]
I did boot with 0x12345678 ig-platform-id and was able to continue the install process (to the configuration part - Country, Account, iCloud, etc.).

I haven't copied anything yet to my SSD:EFI folder which contains only the APPLE folder.

In System Information it says:
Processor 2.9 GHz Intel Core i7 but I have Intel Core i9.
Should I do something to change this?

Should I change ig-platform-id to what its original value: 0x3e9b0000 (before I reboot my laptop)?
Anyhow, for now it indicates wrongly: 
Graphics Intel UHD Graphics 630 (OK) 31 MB (wrong) - it should be 1536 MB.

Display:
It does not say that is a Built-In Retina Display either.

I too have installed an dw1830 card. The thing I still need to do is to purchase an extra antenna to make bleutooth to work. It does not seem to detect anything at the moment but wi-fi works.

Thanks for your help.

**Comment by kevinshroff** [comment]
Just do what Raymonf already told you to do - it will resolve these issues

**Comment by pmtremblay** [comment]
Sorry but it did not resolve any of the issues. 

I tried to change ig-platform-id to what its original value: 0x3e9b0000 but it generates a system panic. I reedit the config.plist to change it back to 0x12345678, then it boots normally except that I have the same list of errors.

Any idea about what could be wrong?

Thanks.

**Comment by Raymonf** [comment]
>I haven't copied anything yet to my SSD:EFI folder which contains only the APPLE folder.

Well, yeah. What I meant was to do that. You could even just make a backup of the current EFI folder and just overwrite it all with what's in this repo.

You can't boot without doing that. There's an issue with the CFL framebuffer kext where it can't determine the number of active lanes; for the UHD 630, you have to patch it to set it to 4 permanently. That _is_ done by the `config.plist` in the EFI folder from the repo.

tl;dr: Yes, overwrite YOUR entire SSD:EFI folder with the one in this repository.

**Comment by pmtremblay** [comment]
Bonjour Raymonf,

What you mean is the EFI folder and not the Install_EFI folder when you're talking repo. If that is the case, I used the EFI folder content already with the "AppleIntelCFLGraphicsFramebuffer KextToPatch present in the config.plist. For some reason, it does not use 1536 MB but only 31 MB!

I tried also to change the g-platform-id to what its original value: 0x3e9b0000 but this does create a kernel panic.

???

Cheers.

**Comment by Xigtun** [comment]
> Thanks. I did that.
> 1- I first start fresh with the content of the folder named: install_EFI onto EFI on my USB drive.
> 2- Put the USB key into my XPS mobile and power it up.
> 3- Press F12 and select the "Installer macOS Mojave" USB drive as my UEFI boot up process.
> 4- I then reformat the SSD drive to Mac OS Extended (Journaled) with GUID Partition Map and name it "MojaveSSD".
> 5- Then I start the Mojave Install Process. Everything works perfectly up to here.
> 6- Then the screen turns black (reboot). This is where I turn off my mobile.
> 7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.
> 8- I put the newly updated USB\EFI key back into my XPS mobile and power it up.
> 9- Press F12 and select the "MojaveSSD" SSD drive as I want to continue the install.
> 10- Then it crashes. I does not continue to install.
> 
> Let me know if I do something wrong or if I missed a step.
> Thanks.
> PM


6- Then the screen turns black (reboot). This is where I turn off my mobile.
6.5 - still use the install efi to boot into the system and rebuild the kextcache replace the efi
7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.

**Comment by Raymonf** [comment]
Are you using 10.14.1? The patch only supports 10.14 GM.

**Comment by pmtremblay** [comment]
10.14 GM. I have not updated to 10.14.1 just yet.

Thanks.

**Comment by Raymonf** [comment]
Hmm.. does Windows show your device as "Intel UHD 630" or is it something else? Could you send over the hardware IDs?

在 2018年10月30日週二 下午9:52，pmtremblay <notifications@github.com<mailto:notifications@github.com>> 寫道：

10.14 GM. I have not updated to 10.14.1 just yet.

Thanks.

**Comment by pmtremblay** [comment]
System Information says:
macOS Mojave
Version 10.14
MacBook Pro (15-inch, 2018)
Processor 2.9 GHz Intel Core i7
Memory 32 GB 2667 MHz DDR4
Startup Disk MojaveSSD
Graphics Intel UHD Graphics 630 31 MB
Serial Number Cxxxxxxx

**Comment by pmtremblay** [comment]
> > Thanks. I did that.
> > 1- I first start fresh with the content of the folder named: install_EFI onto EFI on my USB drive.
> > 2- Put the USB key into my XPS mobile and power it up.
> > 3- Press F12 and select the "Installer macOS Mojave" USB drive as my UEFI boot up process.
> > 4- I then reformat the SSD drive to Mac OS Extended (Journaled) with GUID Partition Map and name it "MojaveSSD".
> > 5- Then I start the Mojave Install Process. Everything works perfectly up to here.
> > 6- Then the screen turns black (reboot). This is where I turn off my mobile.
> > 7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.
> > 8- I put the newly updated USB\EFI key back into my XPS mobile and power it up.
> > 9- Press F12 and select the "MojaveSSD" SSD drive as I want to continue the install.
> > 10- Then it crashes. I does not continue to install.
> > Let me know if I do something wrong or if I missed a step.
> > Thanks.
> > PM
> 
> 6- Then the screen turns black (reboot). This is where I turn off my mobile.
> 6.5 - still use the install efi to boot into the system and rebuild the kextcache replace the efi
> 7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.

Hey Xigtun.

Thanks for paving the way to allow us to use this very good laptop.

I was able to install Mojave (without clearing kext cache) but I still have issues with my Graphics memory.

Any idea how I could solve my issues (Graphics memory and Intel Core i7 instead of i9).

Cheers.

**Comment by Raymonf** [comment]
You're going to need to show the information from Windows. Do you have it installed?

在 2018年10月31日週三 上午11:36，pmtremblay <notifications@github.com<mailto:notifications@github.com> 寫道：

System Information says:
macOS Mojave
Version 10.14
MacBook Pro (15-inch, 2018)
Processor 2.9 GHz Intel Core i7
Memory 32 GB 2667 MHz DDR4
Startup Disk MojaveSSD
Graphics Intel UHD Graphics 630 31 MB
Serial Number Cxxxxxxx

**Comment by pmtremblay** [comment]
No. I don't have Windows installed. I have my original NVMe SSD with Windows on it. I could still swap it in (replace my WD 3D NAND with my original NVMe SSD). If this would still work for you. What Windows application you want me to use in order to give you the information you ask for?

**Comment by Raymonf** [comment]
Device Manager should do it, right click > Properties > Details > Hardware IDs

在 2018年10月31日週三 下午12:19，pmtremblay <notifications@github.com<mailto:notifications@github.com> 寫道：

No. I don't have Windows installed. I have my original NVMe SSD with Windows on it. I could still swap it in (replace my WD 3D NAND with my original NVMe SSD). If this would still work for you. What Windows application you want me to use in order to give you the information you ask for?

**Comment by Xigtun** [comment]
> > > Thanks. I did that.
> > > 1- I first start fresh with the content of the folder named: install_EFI onto EFI on my USB drive.
> > > 2- Put the USB key into my XPS mobile and power it up.
> > > 3- Press F12 and select the "Installer macOS Mojave" USB drive as my UEFI boot up process.
> > > 4- I then reformat the SSD drive to Mac OS Extended (Journaled) with GUID Partition Map and name it "MojaveSSD".
> > > 5- Then I start the Mojave Install Process. Everything works perfectly up to here.
> > > 6- Then the screen turns black (reboot). This is where I turn off my mobile.
> > > 7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.
> > > 8- I put the newly updated USB\EFI key back into my XPS mobile and power it up.
> > > 9- Press F12 and select the "MojaveSSD" SSD drive as I want to continue the install.
> > > 10- Then it crashes. I does not continue to install.
> > > Let me know if I do something wrong or if I missed a step.
> > > Thanks.
> > > PM
> > 
> > 
> > 6- Then the screen turns black (reboot). This is where I turn off my mobile.
> > 6.5 - still use the install efi to boot into the system and rebuild the kextcache replace the efi
> > 7- I take the USB key and stick it into another Mac I have in order to change the content of the USB\EFI with the EFI folder that you supplied.
> 
> Hey Xigtun.
> 
> Thanks for paving the way to allow us to use this very good laptop.
> 
> I was able to install Mojave (without clearing kext cache) but I still have issues with my Graphics memory.
> 
> Any idea how I could solve my issues (Graphics memory and Intel Core i7 instead of i9).
> 
> Cheers.

replace the 4k patch as follow link,
it should worked both 10.14 and 10.14.1 (I have test it on mojave 10.14)
https://www.tonymacx86.com/threads/fix-coffee-lake-intel-uhd-graphics-630-on-macos-mojave-kernel-panic-due-to-divide-by-zero.261687/

**Comment by pmtremblay** [comment]
Bonjour,

I wiped out my UEFI - Windows Boot Manager previously. It was a little hard to go back into Windows 10 Pro. Anyhow...
In Device Manager under "Display adapters" I picked "Intel(R) UHD Graphics 630" and selected (right-click) "Properties" and under the "Details" tab it says: 

Intel(R) UHD Graphics 630
Property 
Hardware Ids

Value
PCI\VEN_8086&DEV_3E9B&SUBSYS_087C1028&REV_00 
PCI\VEN_8086&DEV_3E9B&SUBSYS_087C1028
PCI\VEN_8086&DEV SE9B&CC_030000 
PCI\VEN 8086&DEV_3E9B&CC_0300

Do you need anything else?
Thanks.

**Comment by Raymonf** [comment]
Okay, so it should be alright with the patch in that thread. Have you tried that yet? If no, can you get us a crash log?

Basically, let it kernel panic (reboot) and then boot to the desktop with the ig-platform-id. There will be a "report problem to Apple" dialog - press yes and send everything in "show details". Obviously don't send the actual report to Apple though.

在 2018年10月31日週三 下午9:58，pmtremblay <notifications@github.com<mailto:notifications@github.com> 寫道：

Bonjour,

I wiped out my UEFI - Windows Boot Manager previously. It was a little hard to go back into Windows 10 Pro. Anyhow...
In Device Manager under "Display adapters" I picked "Intel(R) UHD Graphics 630" and selected (right-click) "Properties" and under the "Details" tab it says:

Intel(R) UHD Graphics 630
Property
Hardware Ids

Value
PCI\VEN_8086&DEV_3E9B&SUBSYS_087C1028&REV_00
PCI\VEN_8086&DEV_3E9B&SUBSYS_087C1028
PCI\VEN_8086&DEV SE9B&CC_030000
PCI\VEN 8086&DEV_3E9B&CC_0300

Do you need anything else?
Thanks.

**Comment by pmtremblay** [comment]
I changed the ig-platform-id = 0x3E9B0000 in SSD:EFI\CLOVER\config.plist again and it worked fine this time. It now says: Graphics Intel UHD Graphics 630 2048 MB! Yah baby! Many thanks to you and all the other contributors.
